script.js

```diff
@@ -121,15 +128,42 @@ const markDoneTodoHandler = (
 
     todo.title = sanitizeInput(inputElement.value).trim();
     todo.isCompleted = true;
+    filterTodosHandler(filterValue);
 };
 
 const searchHandler = () => {
     const searchedValue = $searchInput.value.toLowerCase().trim();
-    const searchedArray = todos.filter((todo) =>
+    searchedArray = todos.filter((todo) =>
         todo.title.toLowerCase().includes(searchedValue)
     );
 
-    renderTodos(searchedArray);
+    filterTodosHandler(filterValue);
+};
+
+const filterTodosHandler = (toFilterValue) => {
+    isSearched = $searchInput.value.trim().length ? true : false;
+    let tobeFilteredArray = isSearched ? searchedArray : todos;
+
+    switch (toFilterValue) {
+        case "incomplete":
```

Comment:
Let's use constants to make the project less bug prune.
```suggestion
import { COMPLETE, INCOMPLETE } from "./const.js"

        case "INCOMPLETE":
```